fil_fungible_token/src/token/mod.rs

```diff
@@ -355,16 +352,10 @@ where
         owner: &Address,
         amount: &TokenAmount,
     ) -> Result<TokenAmount> {
-        if amount.is_negative() {
-            return Err(TokenError::InvalidNegative(format!(
-                "burn amount {} cannot be negative",
-                amount
-            )));
-        }
-
         // owner and operator must exist to burn from
         let operator = expect_id(operator)?;
         let owner = self.resolve_to_id(owner)?;
+        let amount = validate_amount(amount, "burn", self.granularity)?;
```

Comment:
is this check purposely after the `resolve_to_id` calls?

in my mind, it's lighter to have the validate_amount check fail first always

Reply:
I moved it to have the param validation match the param order, but you make a good point and I'll move it back.